packages/transport-webtransport/src/muxer.ts

```diff
@@ -0,0 +1,95 @@
+import { webtransportBiDiStreamToStream } from './stream.js'
+import { inertDuplex } from './utils/inert-duplex.js'
+import type WebTransport from './webtransport.js'
+import type { ComponentLogger, Stream, StreamMuxer, StreamMuxerFactory, StreamMuxerInit } from '@libp2p/interface'
+
+export interface WebTransportMuxerInit {
+  maxInboundStreams: number
+}
+
+export function webtransportMuxer (wt: Pick<WebTransport, 'close' | 'createBidirectionalStream'>, reader: ReadableStreamDefaultReader<WebTransportBidirectionalStream>, logger: ComponentLogger, config: WebTransportMuxerInit): StreamMuxerFactory {
+  let streamIDCounter = 0
+  const log = logger.forComponent('libp2p:webtransport:muxer')
+
+  return {
+    protocol: 'webtransport',
+    createStreamMuxer: (init?: StreamMuxerInit): StreamMuxer => {
+      // !TODO handle abort signal when WebTransport supports this.
+
+      if (typeof init === 'function') {
+        // The api docs say that init may be a function
+        init = { onIncomingStream: init }
+      }
+
+      const activeStreams: Stream[] = []
+
+      void Promise.resolve().then(async () => {
+        //! TODO unclear how to add backpressure here?
+        while (true) {
+          const { done, value: wtStream } = await reader.read()
+
+          if (done) {
+            break
+          }
+
+          if (activeStreams.length >= config.maxInboundStreams) {
+            log('too many inbound streams open, closing new incoming stream')
```

Comment:
we should log activeStreams.length and maxInboundStreams

Reply:
Done!